src/handlers/int-test-support/helpers/s3Helper.ts

```diff
@@ -132,6 +136,76 @@ const deleteS3Objects = async (
   );
 };
 
+/* Deletes s3 objects by prefixes in batches */
+export const batchDeleteS3ObjectsByPrefixesInBatchBasic = async (
+  bucketName: string,
+  prefixesToDelete: string[],
+  batchSize: number
+): Promise<DeleteObjectsCommandOutput> => {
+  let result: DeleteObjectsCommandOutput = {
+    Deleted: [],
+    Errors: [],
+    $metadata: {},
+  };
+  for (const prefixToDelete of prefixesToDelete) {
+    let continuationToken: string | undefined;
+
+    do {
+      const listParams = {
+        Bucket: bucketName,
+        Prefix: prefixToDelete,
+        ContinuationToken: continuationToken,
+      };
+      const listResult = await s3Client.send(
```

Comment:
Have you thought about the compatibility of this with the int-test-lambda? Both the `listObjects` call and the `deleteObjects` call need to be made compatible to the calling happening through the lambda. Best to extend/generify the functions for list and delete which are already there.

Reply:
No, thanks added condition runViaLambda in the functions batchDeleteS3ObjectsByKeysInBatch and batchDeleteS3ObjectsByPrefixesInBatch. These functions use DeleteObjectsCommand and ListObjectsV2Command not DeleteObjectCommand/ListObjectsCommand. How can i extend the existing function list and delete?

Reply:
The old `deleteS3Objcts` already takes a bucket and one prefix to delete objects, but does it not in batches and not for more than 1000, so is actually not really a good implementation in the first place. Your new implementation can overwrite the old one, with the old one then becoming a special case of your new one when you just pass one prefix.

Similarly your new implementation of `ListS3Objects` can replace the old one, as the old one also stops at 1000 objects.

Reply:
refactored as suggested